src/algebra/module/linear_map.lean

```diff
@@ -215,6 +234,16 @@ f.to_add_monoid_hom.map_sub x y
 instance : is_add_group_hom f :=
 { map_add := map_add f }
 
+instance compatible_smul.int_module
+  {S : Type*} [semiring S] [has_scalar ℤ S] [semimodule ℤ M]
+  [semimodule S M] [semimodule ℤ M₂] [semimodule S M₂] : compatible_smul M M₂ ℤ S :=
```

Comment:
Does this ever actually happen? If `[semiring S] [has_scalar ℤ S]` then presumably `S` is a ring, and from where I'm sitting it seems unlikely that in practice someone would prove that something is a ring by showing it's a semiring with a scalar action of the integers. Is this instance useful? Is there a situation where this applies but the user either doesn't want or doesn't have access to `[ring S] [add_comm_group M] [module S M]`?

Reply:
I wanted this lemma in https://github.com/leanprover-community/mathlib/pull/5476/files#r548338701.

This does happen - consider a case where I have `add_comm_group M` `semimodule \N M`. R=\N isn't and cannot be a ring.

Sure, I could just require `semimodule \Z M`, but I don't want to do that because I want to use the same base ring for my entire def.

Reply:
I will update this PR to include the lemma it simplifies once #5476 is merged

Reply:
https://github.com/leanprover-community/mathlib/pull/5430/files#r551947248